.github/workflows/clients-java-build.yml

```diff
@@ -0,0 +1,182 @@
+      packages: write
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: github
+        settings-path: ${{ github.workspace }}
+
+    - name: Format linter
+      working-directory: src/clients/java/src/tigerbeetle-java
+      run: mvn -B formatter:validate --file pom.xml
+
+    - name: Install Zig
+      working-directory: src/clients/java
+      run: |
+        git submodule init
+        git submodule update
+        ./scripts/install.sh
+
+    - name: Example programs
+      working-directory: src/clients/java/examples
+      run: mvn compile
+
+    - name: Build
+      if: ${{ github.event_name == 'pull_request' }}
+      working-directory: src/clients/java/src/tigerbeetle-java
+      run: mvn -B package --file pom.xml
+
+    - name: Save local package
+      uses: actions/upload-artifact@v3
+      if: ${{ github.event_name == 'pull_request' }}
+      with:
+        name: jar-artifact
+        path: |
+          src/clients/java/src/tigerbeetle-java/target/*.jar
+          !src/clients/java/src/tigerbeetle-java/target/*javadoc.jar
+
+    - name: Build and publish
+      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+      working-directory: src/clients/java/src/tigerbeetle-java
+      run: mvn -B deploy --file pom.xml -s $GITHUB_WORKSPACE/settings.xml
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  prepare_validation_tests:
+    needs: build
+    runs-on: ubuntu-latest
+    name: Prepare validation tests
+    
+    steps:
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: github
+        settings-path: ${{ github.workspace }}
+
+    ## If it's a pull request, the package wasn't uploaded, so we use a local version:
+    - name: Restore local package
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: actions/download-artifact@v3
+      with:
+        name: jar-artifact
+        path: test/
+
+    ## If it's a push, we can download the package from Maven's registry:
+    - if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+      uses: whelk-io/maven-settings-xml-action@v4
+      with:
+        repositories: '[{ "id": "github", "url": "https://maven.pkg.github.com/tigerbeetledb/tigerbeetle", "snapshots": { "enabled": "true" } }]'
+        servers: '[{ "id": "github", "username": "${{github.actor}}", "password": "${{ github.token }}" }]'
+    - name: Download tigerbeetle-java package
+      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+      run: |
+        mvn dependency:copy -Dartifact=com.tigerbeetle:tigerbeetle-java:0.0.1-SNAPSHOT -DoutputDirectory=. -DrepositoryId=github
+        mkdir -p test
+        mv *.jar test/
+   
+    - name: Create a simple test program
+      run: |
+        mkdir -p test
```

Comment:
Would it make sense to just include this file in a "test" or "examples" directory? That way if you ever want to reproduce what CI is doing, it already exists.